packages/roosterjs-content-model-plugins/lib/autoFormat/AutoFormatPlugin.ts

```diff
@@ -251,10 +264,40 @@ export class AutoFormatPlugin implements EditorPlugin {
                             }
                         );
                     }
+                    break;
+                case 'Enter':
+                    this.handleEnterKey(editor, event);
+                    break;
             }
         }
     }
 
+    private handleEnterKey(editor: IEditor, event: KeyDownEvent) {
+        if (this.options.autoHorizontalLine) {
+            formatTextSegmentBeforeSelectionMarker(editor, (model, _, para, __, context) => {
```

Comment:
Could we move this code to the insertHorizontalLineIntoModel file? I think it would make the file more readable in case we need to add more options in Enter Key.

Reply:
Sounds good, will refactor this soon.

Reply:
Fixed in latest commit, thanks.